javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/FixedSizeQueue.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.muzzle.collector;
+
+import java.util.ArrayDeque;
+import java.util.Iterator;
+
+final class FixedSizeQueue<T> implements Iterable<T> {
```

Comment:
I think because of use of graph API, we already expect to not be able to remove Guava dependency from muzzle. In which case you might use `EvictingQueue`

Reply:
Or just a ring buffer backed by fixed size array?